packages/stateful/actions/components/AuthzExec.tsx

```diff
@@ -0,0 +1,300 @@
+import { Check, Close } from '@mui/icons-material'
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+
+import {
+  CodeMirrorInput,
+  InputErrorMessage,
+  InputLabel,
+  LockWithKeyEmoji,
+  SelectInput,
+  TextInput,
+} from '@dao-dao/stateless'
+import { ActionComponent } from '@dao-dao/types/actions'
+import {
+  validateAddress,
+  validateJSON,
+  validateValidatorAddress,
+} from '@dao-dao/utils'
+
+import { AuthzExecActionTypes } from '../actions/AuthzExec'
+import { ActionCard } from './ActionCard'
+
+export interface AuthzExecOptions {}
+
+export const useAuthzExecActionTypes = (): {
+  type: AuthzExecActionTypes
+  name: string
+}[] => {
+  const { t } = useTranslation()
+
+  return [
+    {
+      type: AuthzExecActionTypes.Delegate,
+      name: t('title.stake'),
+    },
+    {
+      type: AuthzExecActionTypes.Undelegate,
+      name: t('title.unstake'),
+    },
+    {
+      type: AuthzExecActionTypes.Redelegate,
+      name: t('title.restake'),
+    },
+    {
+      type: AuthzExecActionTypes.ClaimRewards,
+      name: t('title.claimRewards'),
+    },
+    {
+      type: AuthzExecActionTypes.Custom,
+      name: t('title.custom'),
+    },
+  ]
+}
+
+export const AuthzExecComponent: ActionComponent<AuthzExecOptions> = (
+  props
+) => {
+  const { t } = useTranslation()
+  const { fieldNamePrefix, onRemove, errors, isCreating } = props
+  const { control, register, watch } = useFormContext()
+
+  const authzExecActions = useAuthzExecActionTypes()
+
+  const authzExecActionType = watch(fieldNamePrefix + 'authzExecActionType')
+
+  return (
+    <ActionCard
+      Icon={LockWithKeyEmoji}
+      onRemove={onRemove}
+      title={t('title.authzExec')}
+    >
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel
+          name={t('form.authzExecActionType')}
+          tooltip={t('form.authzExecActionTypeTooltip')}
+        />
+        <SelectInput
+          containerClassName="grow"
+          defaultValue={authzExecActions[0].type}
+          disabled={!isCreating}
+          error={errors?.authzExecActionType}
+          fieldName={fieldNamePrefix + 'authzExecActionType'}
+          register={register}
+        >
+          {authzExecActions.map(({ name, type }, idx) => (
+            <option key={idx} value={type}>
+              {name}
+            </option>
+          ))}
+        </SelectInput>
+      </div>
+
+      {authzExecActionType === AuthzExecActionTypes.Delegate && (
+        <>
+          <div className="flex flex-col items-stretch gap-1">
+            <InputLabel name={t('form.delegatorAddress')} />
+            <TextInput
```

Comment:
`AddressInput` is better suited for wallet addresses, and if you pass in the stateful `ProfileDisplay` component to the `ProfileDisplay` prop of `AddressInput`, it'll display their profile!